### PR TITLE
pytest migration - 1

### PR DESCRIPTION
**Pull request by SaitejaUtpala**

### CODE REVIEW

File: geomstats/tests.py [hunk 1]
```diff
@@ -31,101 +31,55 @@ def tf_backend():
     return os.environ["GEOMSTATS_BACKEND"] == "tensorflow"
 
 
-def autograd_only(test_item):
-    """Decorate to filter tests for autograd only."""
-    if autograd_backend():
-        return test_item
-    return unittest.skip("Test for autograd backend only.")(test_item)
+autograd_only = pytest.mark.skipif(
+    not autograd_backend, reason="Test for autograd backend only."
```
Comment (ninamiolane) [inline geomstats/tests.py:35]: should this be autograd_backend()?

Comment (ninamiolane) [reply geomstats/tests.py:35]: same for below

File: geomstats/tests.py [hunk 2]
```diff
@@ -31,101 +31,55 @@ def tf_backend():
...
+np_autograd_and_tf_only = pytest.mark.skipif(
+    not (np_backend() or autograd_backend() or tf_backend()),
+    reason="Test for numpy, autograd and tensorflow backends only.",
+)
+np_autograd_and_torch_only = pytest.mark.skipif(
+    not (np_backend() or autograd_backend() or pytorch_backend()),
+    reason="Test for numpy, autograd and pytorch backends only.",
+)
+autograd_tf_and_torch_only = pytest.mark.skipif(
+    np_backend(), reason="Test for backends with automatic differentiation only"
```
Comment (ninamiolane) [inline geomstats/tests.py:75]: Mssing .

File: geomstats/tests.py [hunk 3]
```diff
@@ -31,101 +31,55 @@ def tf_backend():
...
+    not (np_backend() or autograd_backend()),
+    reason="Test for numpy and autograd backends only.",
+)
+autograd_and_torch_only = pytest.mark.skipif(
+    not (autograd_backend() or pytorch_backend()),
+    reason="Test for numpy and autograd backends only.",
+)
+autograd_and_tf_only = pytest.mark.skipif(
+    not (autograd_backend() or tf_backend()),
+    reason="Test for numpy and autograd backends only.",
```
Comment (ninamiolane) [inline geomstats/tests.py:63]: Change reason

File: geomstats/tests.py [hunk 4]
```diff
@@ -31,101 +31,55 @@ def tf_backend():
...
+    not (np_backend() or pytorch_backend()),
+    reason="Test for numpy and pytorch backends only.",
+)
+np_and_autograd_only = pytest.mark.skipif(
+    not (np_backend() or autograd_backend()),
+    reason="Test for numpy and autograd backends only.",
+)
+autograd_and_torch_only = pytest.mark.skipif(
+    not (autograd_backend() or pytorch_backend()),
+    reason="Test for numpy and autograd backends only.",
```
Comment (ninamiolane) [inline geomstats/tests.py:59]: Change reason

File: geomstats/tests.py [hunk 5]
```diff
@@ -155,19 +109,21 @@ def assertAllClose(self, a, b, rtol=gs.rtol, atol=gs.atol):
...
+    def assertEqual(self, a, b):
+        assert a == b
+
     def assertAllCloseToNp(self, a, np_a, rtol=gs.rtol, atol=gs.atol):
         are_same_shape = np.all(a.shape == np_a.shape)
         are_same = np.allclose(a, np_a, rtol=rtol, atol=atol)
-        if tf_backend():
-            return super().assertTrue(are_same_shape and are_same)
-        return super().assertTrue(are_same_shape and are_same)
+        assert are_same and are_same_shape == True
```
Comment (ninamiolane) [inline geomstats/tests.py:124]: I don't think you need == True at the end here

File: tests/tests_geomstats/test__my_manifold.py [hunk 6]
```diff
@@ -46,7 +46,7 @@ class TestMyManifold(geomstats.tests.TestCase):
     as in the examples below.
     """
 
-    def setUp(self):
+    def setup_method(self):
```
Comment (ninamiolane) [inline tests/tests_geomstats/test__my_manifold.py:49]: Oh! pytest renames all of these?

File: tests/tests_geomstats/test_agglomerative_hierarchical_clustering.py [hunk 7]
```diff
@@ -9,14 +9,14 @@
 )
 
 
+@geomstats.tests.np_and_autograd_only
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_agglomerative_hierarchical_clustering.py:12]: Nice!

File: tests/tests_geomstats/test_backends.py [hunk 8]
```diff
@@ -790,7 +789,7 @@ def test_choice(self):
             result_bool = False
 
         self.assertTrue(result_bool)
-        self.assertEqual(len(result), a)
+        assert len(result) == a
```
Comment (ninamiolane) [inline tests/tests_geomstats/test_backends.py:792]: NIT: why not self.assertEqual(len(result), a)?

**Review by ninamiolane — CHANGES_REQUESTED** [review]
Very nice! Thanks for doing this. I've put a first batch of (mostly minor) comments.

Could you update the PR title too? 🙌  thank you!

File: tests/tests_geomstats/test_backends.py (hunk 8 above)
Comment (SaitejaUtpala) [reply tests/tests_geomstats/test_backends.py:792]: yup.. first I thought why equal...  Then I was experimenting.

File: geomstats/tests.py (hunk 5 above)
Comment (SaitejaUtpala) [reply geomstats/tests.py:124]: correct

**Pull request merged**